### ISSUE TITLE
Only STweep a highlighted mouse selection

### ISSUE THREAD

**Issue by HAHermsen** [Barteling/STweep#14]
Hi,

I'd like to see a feature which lets it only STweep a highlighted selection of text instead of the entire code or declaration tab.
This enables users to Stweep arbitrary number lines of selected text of a single tab only.

Thank you

**Comment by RogerChristopher** [comment]
I thought this, but the best use of STweep is to apply it to the whole project to enforce a consistent style amongst multiple users.

Given that it may be a while before all the formatting use cases are implemented, a better option might be to be able to mark a section of code that should be ignored by STweep by using a custom attribute. This would allow custom formatting of code where STweep does not yet 'do the right thing', but still apply STweep to the whole project.

**Comment by HAHermsen** [comment]
All true, but also please consider that not everybody works in a team or behind a CI/CD system with a webhooks setup.
Like me, I don't use a full fledged CI/CD system for my code. That would be overkill. I think that STweep can cater to small human interaction up to full fledged automated experience.

**Comment by Roald87** [comment]
I agree with @RogerChristopher, that using a custom attribute would be a better idea. Other alternatives could be a region, or comment.

**Comment by HAHermsen** [comment]
See updated request above

**Comment by Barteling** [comment]
Partial formatting is on the to do list. It has been requested more.

I believe that formatting the active selection as well as skipping marked regions is useful.
For skipping regions I think using a comment might be the most practical way to go:
//STweep disable
 ignored code
//STweep enable  or and of file..

**Comment by I-Campbell** [comment]
Comments should not effect code (particularly as noone reads them anyway ;p).
I would prefer a [custom pragma](https://help.codesys.com/webapp/_cds_f_pragmas_attribute;product=codesys;version=3.5.16.0).
```
{attribute 'stweep_off'}
But(   myCode:='already lovely');
{attribute 'stweep_on'}
```

**Comment by Barteling** [comment]
I'm working on this feature, I hope to have it in a beta soon so stay tuned for that.

**Comment by Barteling** [comment]
Stweep 3.5.1-Beta for Twincat and Codesys has support for disabling and enabling STweep.
This can be done with either {STweep.Disable} , {STweep.Enable}  or (* STweep.Disable * ), (** STweep.Enable *). Or mixed.
https://www.stweep.com/beta-releases

Looking forward to any feedback regarding this feature.

**Comment by masterprinter** [comment]
The pragmas {STweep.Disable}  {STweep.Enable} work as intended.
But a minor bug is that {Stweep.Disable} is auto corrected to {Stweep.disable} and {Stweep.Enable} isnt.
Tested in VS2019 with STweep formatter for TwinCAT 4.1.2
